### DISCUSSION TITLE
Allow sorting Top Sources by each of the three categories

### DISCUSSION THREAD

**Discussion by preslavrachev** [plausible/analytics#458] [Anything Else]
## Prerequisites
- [x] I have searched open and closed issues to make sure that the feature has not yet been requested.

## Feature request
At the moment, the full list of top sources is ordered by the number of visitors. It would be very helpful to make it possible to sort them based on any of the remaining categories (e.g. Bounce Rate and Visit Duration)

![image](https://user-images.githubusercontent.com/86060/101527221-01024f80-398e-11eb-925d-cb7655467d96.png)

**How would you like it to work**
By default, Top Sources will remain sorted by the number of visitors, but the column will have a sort icon, suggesting it is the one that results have been sorted upon. All column headers will be clickable. By clicking on any of the remaining columns, it will become the actively sorted upon and the list of results will be re-rendered to reflect that 

---
**NOTE:** As usually, If there is enough interest in getting this feature implemented, I can implement it myself and send a PR. 

**Update:** Sorting is now possible in the Details view of any report. Just click on the metric you'd like to sort by. You can sort high to low or low to high.

**Comment by ukutaht** [comment]
Agreed, I think this would be a great addition

**Comment by metmarkosaric** [comment]
Would be great for Top Pages too!

**Comment by preslavrachev** [comment]
Great! I’ll set some time aside at the start of the next week to do it.

**Comment by preslavrachev** [comment]
Started working on it. The change will require adding a `sort` attribute to each stats `Plausible.Stats.Query`. It will be used when present, or will default to "clicks" as it is at the moment.

**Reply by ukutaht** [reply]
Makes sense, let me know if you have any questions

**Reply by preslavrachev** [reply]
Something like this is what I'm having in mind: https://github.com/plausible/analytics/compare/master...preslavrachev:issue-458-allow-sorting-of-top-sources?expand=1 It's not really DRY, but I'd say let's see if it sticks, and we can definitely think of replicating it in other places using a query.

I still need to wire up the `sort` parameter through the API controller and bind with the frontend.

**Comment by OlayaGonzalez** [comment]
Hi guys! I was just wondering what's the status of this. Would be amazing to be able to sort by Pageviews as well when looking at Top Pages (unfiltered and filtered).

**Comment by metmarkosaric** [comment]
Sorting is now possible in the Details view of any report. Just click on the metric you'd like to sort by. You can sort high to low or low to high.
